### PR TITLE
Fix host using localhost not working on a lot of envs

### PR DESCRIPTION
**Pull request by alexandrebodin**
#### Description of what you did:

Fix host using lcoalhost not working on a lot of envs. I also added default env vars usage in all envs. 

Fix https://github.com/strapi/strapi-docker/issues/191
Fix https://github.com/strapi/strapi/issues/5652

### CODE REVIEW

File: examples/getstarted/config/environments/development/server.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "host": "localhost",
-  "port": 1337,
+  "host": "${process.env.HOST || '0.0.0.0'}",
```
Comment (abdonrd) [inline examples/getstarted/config/environments/development/server.json]: That's right?

Screenshot 2020-03-31 at 15 39 24

Comment (alexandrebodin) [reply examples/getstarted/config/environments/development/server.json]: @abdonrd  what is your question ?

Comment (abdonrd) [reply examples/getstarted/config/environments/development/server.json]: Use the `0.0.0.0` in the development env. Previously show `localhost`.

Comment (alexandrebodin) [reply examples/getstarted/config/environments/development/server.json]: That's what I wanted indeed. the url isn't cool to read that's for sure :/ 
 but it's the only way to make it work with docker for example.

**Comment by fl034** [comment]
This didn't fix it for me. I had to get active and change `localhost` to `0.0.0.0` in `server.json`, like the workaround for 19.4 stated above.
It was working with `localhost` in 19.3 and still broken in 19.5.

I'm using a docker-compose.yml like this (actually with nginx proxy, but I removed it because it doesn't matter in this case):

```
version: '3'

services:
  api:
    image: strapi/strapi:3.0.0-beta.19.5
    environment:
      - DATABASE_CLIENT=mongo
      - DATABASE_HOST=db
      - DATABASE_PORT=27017
      - DATABASE_NAME=strapi
      - DATABASE_USERNAME=
      - DATABASE_PASSWORD=
      - DATABASE_SSL=false
    ports:
      - 1337:1337
    volumes:
      - ./strapi-app:/srv/app
    depends_on:
      - db
    restart: always
  db:
    image: mongo
    environment:
      - MONGO_INITDB_DATABASE=strapi
    volumes:
      - ./volumes/db:/data/db
      - ./volumes/db-dump:/dump
    restart: always
```

**Comment by alexandrebodin** [comment]
The new image is not changing your project locally @fl034 :) When upgrading you still need to use 0.0.0.0 in docker that's the way it should be. 


You can read through this to understand why https://github.com/strapi/strapi-docker/issues/191

**Pull request merged**